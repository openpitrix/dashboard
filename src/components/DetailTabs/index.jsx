import React, { Component } from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';
import { noop } from 'lodash';
import { translate } from 'react-i18next';

import styles from './index.scss';

@translate()
export default class DetailTabs extends Component {
  static propTypes = {
    changeTab: PropTypes.func,
    defaultTab: PropTypes.string,
    tabs: PropTypes.array
  };

  static defaultProps = {
    changeTab: noop,
    tabs: [],
    defaultTab: ''
  };

  constructor(props) {
    super(props);

    this.state = {
      curTab: props.defaultTab || props.tabs[0]
    };
  }

  componentDidMount() {
    const { curTab } = this.state;
    this.props.changeTab(curTab);
  }

  shouldComponentUpdate(nextProps, nextState) {
    return nextState.curTab !== this.state.curTab;
  }

  handleChange = tab => {
    this.setState({ curTab: tab });
  };

  componentDidUpdate() {
    this.props.changeTab(this.state.curTab);
  }

  render() {
    const { tabs, t } = this.props;
    const { curTab } = this.state;

    return (
      <div className={styles.detailTabs}>
        {tabs.map((tab, idx) => (
          <div
            className={classnames(styles.tab, {
              [styles.active]: tab === curTab
            })}
            key={idx}
            onClick={() => this.handleChange(tab)}
          >
            {t(tab)}
          </div>
        ))}
      </div>
    );
  }
}
