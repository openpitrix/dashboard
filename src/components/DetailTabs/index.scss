@import '~scss/vars';

.detailTabs {
  padding: 0 24px;
  height: 47px;
  background: $N0;
  border-bottom: 1px solid $N10;

  >label {
    display: inline-block;
    padding: 15px 0;
    margin-right: 24px;
    line-height: 16px;
    font-size: 14px;
    font-weight: 500;
    color: $N300;
    cursor: pointer;

    &.active {
      color: $P75;
      border-bottom: 1px solid $P75;
    }

    &.disabled {
      display: none;
      color: $N100;
      opacity: 0.5;
      cursor: not-allowed;
    }
  }
}

.accountTabs {
  padding: 0;
  height: 41px;
  background-color: $N500;
  border-bottom: 0 none;

  >label {
    padding: 12px 0;
    color: $N30;

    &.active {
      color: $N0;
      border-color:$Y200;
    }
  }
}
