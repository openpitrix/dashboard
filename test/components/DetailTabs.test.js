
import DetailTabs from 'components/DetailTabs';

describe('Component DetailTabs ', () => {
  it('basic render', () => {
    const wrapper = render(<DetailTabs />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
