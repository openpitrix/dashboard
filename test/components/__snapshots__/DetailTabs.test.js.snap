// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component DetailTabs  basic render 1`] = `
<div
  class="detailTabs"
/>
`;
